pkg/tfshim/sdk-v2/provider.go

```diff
@@ -165,7 +164,18 @@ func (p v2Provider) ReadDataDiff(
 	if !ok {
 		return nil, fmt.Errorf("unknown resource %v", t)
 	}
-	diff, err := r.Diff(ctx, nil, configFromShim(c), p.tf.Meta())
+
+	providerOpts, err := getProviderOptions(p.opts)
+	if err != nil {
+		return nil, err
+	}
+	config := configFromShim(c)
+	rawConfig := makeResourceRawConfig(providerOpts.diffStrategy, config, r)
```

Comment:
Perhaps commit to ```recoverAndCoerceCtyValue(resource, config.Raw)``` for new code? It's stricter, it doesn't produce plausible but invalid garbage. I understand there's risks with handling an error from that, possibly need to start with rolling out under warning handling of the error?